api/nodes/views.py

```diff
@@ -106,7 +115,10 @@ def perform_destroy(self, instance):
         user = self.request.user
         auth = Auth(user)
         node = self.get_object()
-        node.remove_node(auth=auth)
+        try:
+            node.remove_node(auth=auth)
+        except NodeStateError as err:
```

Comment:
Totally agree that yes, a loud error should get raised if this happens. Just wanted to be clear this was intentional.